src/walle-lib/element.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+#include "vector2.hpp"
+
+class Environment;
+
+class Element {
+    public:
+        friend class Environment; // Environment needs to access the protected members of Element
```

Comment:
It is probably a better idea to add what is called accessors (get/set method) instead of allowing direct access to member. The extra level of indirection allowed by accessors will ease software evoluation.

Reply:
I chose not to use setter methods because it seems unwise to allow the x and y position of the element to be modified by anyone, anywhere.
Do you think, despite this, that setters and getters could be a good compromise between modularity and security ?

Reply:
I finally understand that making ```Environment``` a friend of ```Element``` is not really secure, as it allows ```Environment``` to read and write all private attributes. Moreover, using setters and getters ensures better encapsulation of the class behavior. 
[This commit](https://github.com/damienmarchal/junia-2024/pull/12/commits/018681e86cf2ebbadce9c4f960ef007eecf3e0b8) introduced this change.